lib/pure/results.nim

```diff
@@ -0,0 +1,124 @@
+import typetraits, strutils
+
+type
+  Result*[T] = object
+    ## Either a value or an error.
+    case isSuccess*: bool:
+    of true:
+      value*: T
+    of false:
+      error*: ref Exception
+      backtrace*: BacktraceFrame
```

Comment:
Not yet, but it can be done, I already have working version in reactor.nim. It is set in implementation of `await` ([awaitInIterator](https://github.com/zielmicha/reactor.nim/blob/09d7930ff2bbf802604fcea7b01e922f7e242b9c/reactor/async/asyncmacro.nim#L32)).